src/Orleans.Core/Caching/Internal/ConcurrentDictionarySize.cs

```diff
@@ -0,0 +1,220 @@
+using System;
+using System.Collections.Generic;
+
+namespace Orleans.Caching.Internal;
+
+// Derived from BitFaster.Caching by Alex Peck
+// https://github.com/bitfaster/BitFaster.Caching/blob/5b2d64a1afcc251787fbe231c6967a62820fc93c/BitFaster.Caching/ConcurrentDictionarySize.cs
+internal static class ConcurrentDictionarySize
```

Comment:
This could be skippable - from memory, after .NET6 there was a fix to improve the internals of concurrent dictionary such that a sensible prime number is selected. If Orleans only runs on newer frameworks the only benefit of this is lower up-front allocation (if the cache never grows large, a big array will not be allocated - this is only important if you have many caches and you don't know which ones will be hot).